### PR TITLE
Qa sdk aws eg roll apis

### PR DESCRIPTION
**Pull request by chaplabansi**

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/enums/AwsElastiGroupActionTypeEnum.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package com.spotinst.sdkjava.enums;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum AwsElastiGroupActionTypeEnum {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AwsElastiGroupActionTypeEnum.java:7]: ElastiGroup to be changed to Elastigroup as per naming convention.

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetGroupDeploymentStatusResponse.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+
+public class ApiElastigroupGetGroupDeploymentStatusResponse implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetGroupDeploymentStatusResponse.java:17]: Naming convention of this class to be reviewed.

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStartDeploymentOnFailure.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+
+public class ApiElastigroupStartDeploymentOnFailure implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStartDeploymentOnFailure.java:13]: Change class name to ApiElastigroupDeploymentStrategyOnFailure

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 4]
```diff
@@ -1216,6 +1212,175 @@ private static ApiScalingAction toDal(ScalingAction scalingAction) {
...
+
+            if (elastigroupStrategy.isOnFailureSet()) {
+                retVal.setOnFailure(toDal(elastigroupStrategy.getOnFailure()));
+            }
+        }
+
+        return retVal;
+    }
+
+    private static ApiElastigroupStartDeploymentOnFailure toDal(ElastigroupStartDeploymentOnFailure elastigroupOnFailure) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:1268]: rename input parameter to strategyOnFailure

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java [hunk 5]
```diff
@@ -2504,5 +2665,353 @@ private static PredictiveScale toBl(ApiPredictiveScale predictiveScale) {
...
+            if (elastigroupStrategy.isOnFailureSet()) {
+                retVal.setOnFailure(toBl(elastigroupStrategy.getOnFailure()));
+            }
+
+        }
+
+        return retVal;
+    }
+
+    private static ElastigroupStartDeploymentOnFailure toBl(ApiElastigroupStartDeploymentOnFailure elastigroupOnFailure) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:2722]: rename input parameter to strategyOnFailure

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstances.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupGetDeploymentStatusInstances implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstances.java:16]: Remove Get from classname

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesBlue.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupGetDeploymentStatusInstancesBlue implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesBlue.java:15]: Remove Get from classname

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesGreen.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupGetDeploymentStatusInstancesGreen implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesGreen.java:15]: Remove Get from classname

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstances.java [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupGetDeploymentStatusInstances {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstances.java:13]: Remove Get from classname

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesBlue.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupGetDeploymentStatusInstancesBlue {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesBlue.java:12]: Remove Get from classname

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesGreen.java [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupGetDeploymentStatusInstancesGreen {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesGreen.java:12]: Remove Get from classname

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusResponse.java [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupGetDeploymentStatusResponse implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                                      isSet;
+    private ApiElastigroupDeploymentProgress                 progress;
+    private Integer                                          numOfBatches;
+    private Integer                                          currentBatch;
+    private Integer                                          gracePeriod;
+    private String                                           strategyAction;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusResponse.java:24]: Is this not Enum?

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeployment.java [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.Set;
+
+public class ElastigroupStartDeployment {
+    @JsonIgnore
+    private Set<String>                        isSet;
+    private Integer                            batchSizePercentage;
+    private Integer                            drainingTimeout;
+    private Integer                            gracePeriod;
+    private AwsElastigroupHealthCheckTypeEnum  healthCheckType;
+    private ElastigroupStartDeploymentStrategy strategy;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeployment.java:16]: This class is not mentioned in the imports?

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentOnFailure.java [hunk 14]
```diff
@@ -0,0 +1,120 @@
+package com.spotinst.sdkjava.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.spotinst.sdkjava.enums.AwsElastiGroupActionTypeEnum;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ElastigroupStartDeploymentOnFailure {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentOnFailure.java:9]: Rename class to ElastigroupStrategyOnFailure

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStrategy.java [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+
+public class ApiElastigroupStrategy implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStrategy.java:13]: Rename to ApiElastigroupDeploymentStrategy

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentStrategy.java [hunk 16]
```diff
@@ -0,0 +1,99 @@
+package com.spotinst.sdkjava.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.spotinst.sdkjava.enums.AwsElastigroupActionEnum;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ElastigroupStartDeploymentStrategy {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentStrategy.java:8]: Rename to ElastigroupDeploymentStrategy

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStopDeploymentRequest.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.spotinst.sdkjava.client.rest.JsonMapper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ElastigroupStopDeploymentRequest {
+
+    //region Members
+    @JsonProperty("roll")
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupStopDeploymentRequest.java:12]: Need to understand more on this?

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStopDeploymentRoll.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+
+public class ApiElastigroupStopDeploymentRoll implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStopDeploymentRoll.java:13]: Rename to ApiElastigroupDeploymentRoll

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStopDeploymentRoll.java [hunk 19]
```diff
@@ -0,0 +1,55 @@
+package com.spotinst.sdkjava.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ElastigroupStopDeploymentRoll {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupStopDeploymentRoll.java:8]: ElastigroupDeploymentRoll

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java [hunk 20]
```diff
@@ -676,4 +676,127 @@ public Boolean removeSuspendedScalingPolicies(
...
+
+        //ElastigroupStopDeploymentRequest elastigroupStopDeploymentRequest = stopDeploymentRequest.getStopDeployment();
+
+        RepoGenericResponse<ElastigroupStopDeploymentResponse> stopDeploymentResponse =
+                getSpotinstElastigroupRepo().stopDeployment(stopDeploymentRequest, elastigroupId , deploymentId , authToken, account);
+
+        if(stopDeploymentResponse.isRequestSucceed()){
+            isDeploymentStopped = stopDeploymentResponse.getValue();
+        }
+
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:717]: Remove Linebreak.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java [hunk 21]
```diff
@@ -676,4 +676,127 @@ public Boolean removeSuspendedScalingPolicies(
         }
         return retVal;
     }
-}
+
+    public ElastigroupStartDeploymentResponse startDeployment(ElastigroupStartDeploymentRequest startDeploymentRequest, String elastigroupId) {
+
+        ElastigroupStartDeploymentResponse isDeploymentStarted;
+
+        ElastigroupStartDeployment elastigroupStartDeploymentRequest = startDeploymentRequest.getElastigroupDeployment();
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:684]: Didn't understand this line.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java [hunk 22]
```diff
@@ -676,4 +676,127 @@ public Boolean removeSuspendedScalingPolicies(
...
+                getSpotinstElastigroupRepo().getGroupDeploymentStatus(elastigroupId, authToken, account);
+
+        if(getGroupDeploymentStatusResponse.isRequestSucceed()){
+            getGroupDeploymentResponse =getGroupDeploymentStatusResponse.getValue();
+        }
+        else {
+            List<HttpError> httpExceptions = getGroupDeploymentStatusResponse.getHttpExceptions();
+            HttpError       httpException  = httpExceptions.get(0);
+            LOGGER.error(String.format(
+                    "Error encountered while attempting to get deployment status. Code: %s. Message: %s.",
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:769]: group deployment - please correct message.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java [hunk 23]
```diff
@@ -53,7 +53,7 @@ public static ApiElastigroup createElastigroup(ApiElastigroup groupToCreate, Str
         String uri = String.format("%s/aws/ec2/group", apiEndpoint);
 
         // Send the request
-        RestResponse response = RestClient.sendPost(uri, body, headers, queryParams);
+        RestResponse response = RestClient.sendPut(uri, body, headers, queryParams);
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:56]: Why this is changed to put? Was it changed just for testing.

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
common comment - please use new folder structure for API classes/Requests and Response classes.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java (hunk 23 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java:56]: yes, corrected.

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStartDeploymentOnFailure.java (hunk 3 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStartDeploymentOnFailure.java:13]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStopDeploymentRoll.java (hunk 18 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStopDeploymentRoll.java:13]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentStrategy.java (hunk 16 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentStrategy.java:8]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentOnFailure.java (hunk 14 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupStartDeploymentOnFailure.java:9]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStrategy.java (hunk 15 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupStrategy.java:13]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetGroupDeploymentStatusResponse.java (hunk 2 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetGroupDeploymentStatusResponse.java:17]: Changed to ApiElastigroupGroupDeploymentStatusResponse

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 4 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:1268]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java (hunk 5 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupConverter.java:2722]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstances.java (hunk 6 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstances.java:16]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesGreen.java (hunk 8 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesGreen.java:15]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesBlue.java (hunk 10 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesBlue.java:12]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesGreen.java (hunk 11 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstancesGreen.java:12]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesBlue.java (hunk 7 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusInstancesBlue.java]: Done

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupStopDeploymentRequest.java (hunk 17 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupStopDeploymentRequest.java:12]: Removed roll property, As it's object in the request.

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstances.java (hunk 9 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ElastigroupGetDeploymentStatusInstances.java]: Done

File: src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusResponse.java (hunk 12 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/ApiElastigroupGetDeploymentStatusResponse.java]: yes modified. DOne

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java (hunk 21 above)
Comment (chaplabansi) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClient.java:684]: It is used to get Request Body for startDeloyment.

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**